### PR TITLE
Set thread information earlier in exec_env creation

### PR DESCRIPTION
**Pull request by sjamesr**

### CODE REVIEW

**Comment by sjamesr** [comment]
This tries to address https://github.com/bytecodealliance/wasm-micro-runtime/issues/3966.

File: core/iwasm/common/wasm_exec_env.c [hunk 1]
```diff
@@ -60,6 +60,14 @@ wasm_exec_env_create_internal(struct WASMModuleInstanceCommon *module_inst,
...
 
 #ifdef OS_ENABLE_HW_BOUND_CHECK
+    if (!os_thread_signal_inited()) {
+        goto fail5;
+    }
+
+    /* Set thread handle and stack boundary if they haven't been set */
+    wasm_exec_env_set_thread_info(exec_env);
+    wasm_runtime_set_exec_env_tls(exec_env);
+
```
[the remark this answers is not in the document]
Comment (sjamesr) [reply core/iwasm/common/wasm_exec_env.c]: I moved the code into here, but I placed it after the `sigaltstack` call. Does this look OK to you?

**Review by wenyongh — APPROVED** [review]

File: core/iwasm/common/wasm_exec_env.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (wenyongh) [reply core/iwasm/common/wasm_exec_env.c]: Yes, it is good to me.

**Review by xujuntwt95329 — APPROVED** [review]
LGTM

**Pull request merged**